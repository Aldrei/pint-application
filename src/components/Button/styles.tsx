import styled from 'styled-components'

export const ButtonStyle = styled.button`
  margin-bottom: 10px;
  height: 40px;
  min-height: 40px;
  padding: 0 20px;
  border: 1px solid rgba(255, 255, 255, 0.23);
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  cursor: pointer;

  &:last-child {
    margin-bottom: 0;
  }

  ${props => props.color === 'blue' && `
    background-color: rgb(0, 127, 255);
    color: rgb(255, 255, 255);
  `}

  ${props => props.disabled && `
    cursor: not-allowed;
    opacity: 0.45;
  `}
`