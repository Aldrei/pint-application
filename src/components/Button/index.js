import React from 'react'

import {ButtonStyled} from './styles'

const Button = ({...props}) => <ButtonStyled {...props}>Text</ButtonStyled>

export default Button