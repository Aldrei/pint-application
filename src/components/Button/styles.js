import styled from '@emotion/styled'
import { css } from '@emotion/css'

import Button from '@mui/material/Button'

export const ButtonStyled = styled(Button)`
  padding: 30px !important
  ${css`
    padding: 30px !important
  `}
`