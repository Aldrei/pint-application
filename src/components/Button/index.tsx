import React from 'react'

import { ButtonStyle } from './styles'

const Button = ({...props}): React.ReactElement => {
  return <ButtonStyle data-testid="buttonComp" className="buttonComp" {...props} />
}

export default Button;