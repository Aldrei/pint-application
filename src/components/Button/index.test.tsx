import { render } from '@testing-library/react';
import React from 'react';

import Button from './index';

describe('Button component', () => {
  it('Should render correctly', () => {
    const nodeEl = render(<Button text='Button test' />);

    expect(nodeEl.getByTestId('buttonComp')).toHaveTextContent(`Button test`);
    expect(nodeEl.getByTestId('buttonComp')).not.toBeDisabled();
    
    expect(nodeEl.baseElement).toMatchSnapshot();
  })

  it('Should be disabled', () => {
    const nodeEl = render(<Button text='Test' disabled />);

    expect(nodeEl.getByTestId('buttonComp')).toBeDisabled();

    expect(nodeEl.baseElement).toMatchSnapshot();
  })
})