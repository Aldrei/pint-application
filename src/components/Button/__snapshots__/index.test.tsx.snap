// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should be disabled 1`] = `
<body>
  <div>
    <button
      class="sc-bczRLJ heDyFY buttonComp"
      data-testid="buttonComp"
      disabled=""
    >
      Test
    </button>
  </div>
</body>
`;

exports[`Button component Should render correctly 1`] = `
<body>
  <div>
    <button
      class="sc-bczRLJ hvoMIW buttonComp"
      data-testid="buttonComp"
    >
      Button test
    </button>
  </div>
</body>
`;
